src/_gettsim/visualization.py

```diff
@@ -67,41 +73,66 @@ def plot_dag(
         a hover information. Sometimes, the tooltip is not properly displayed.
 
     """
+    targets = build_targets_tree(DEFAULT_TARGETS if targets is None else targets)
+
+    if isinstance(columns_overriding_functions, dict):
+        names_of_columns_overriding_functions = tree_to_dict_with_qualified_name(
+            columns_overriding_functions
+        ).keys()
+    elif isinstance(columns_overriding_functions, str):
+        names_of_columns_overriding_functions = [columns_overriding_functions]
+    elif columns_overriding_functions is None:
+        names_of_columns_overriding_functions = []
+    else:
+        names_of_columns_overriding_functions = columns_overriding_functions
 
-    targets = DEFAULT_TARGETS if targets is None else targets
-    targets = parse_to_list_of_strings(targets, "targets")
-    columns_overriding_functions = parse_to_list_of_strings(
-        columns_overriding_functions, "columns_overriding_functions"
+    # Load functions.
+    all_functions = add_derived_functions_to_functions_tree(
+        environment=environment,
+        targets=targets,
+        names_of_columns_in_data=names_of_columns_overriding_functions,
+    )
+    functions_not_overridden, _ = _filter_tree_by_name_list(
+        tree=all_functions,
+        qualified_names_list=names_of_columns_overriding_functions,
     )
 
-    # Load functions.
-    functions_not_overridden, functions_overridden = (
-        check_functions_and_differentiate_types(
-            environment,
-            targets=targets,
-            data_cols=list(TYPES_INPUT_VARIABLES),
-        )
+    # Create parameter input structure.
+    input_structure = dags.dag_tree.create_input_structure_tree(
+        functions=functions_not_overridden,
+        targets=None,  # None because no functions should be filtered out
     )
 
     # Select necessary nodes by creating a preliminary DAG.
-    nodes = set_up_dag(
+    dag = set_up_dag(
         all_functions=functions_not_overridden,
         targets=targets,
-        columns_overriding_functions=columns_overriding_functions,
+        names_of_columns_overriding_functions=names_of_columns_overriding_functions,
+        input_structure=input_structure,
         check_minimal_specification=check_minimal_specification,
-    ).nodes
-    necessary_functions = {
-        f_name: f for f_name, f in functions_not_overridden.items() if (f_name in nodes)
-    }
+    )
+
+    _, necessary_functions = _filter_tree_by_name_list(
+        functions_not_overridden, dag.nodes
+    )
+    processed_functions = _round_and_partial_parameters_to_functions(
+        necessary_functions,
```

Comment:
```suggestion
        _filter_tree_by_name_list(...)[1],
```
and remove the above line.